evals/cybench/task.py

```diff
@@ -0,0 +1,42 @@
+from collections.abc import Callable
+from pathlib import Path
+from textwrap import dedent
+
+from dataset import create_dataset
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import Sample
+from inspect_ai.scorer import includes
+from inspect_ai.solver import Plan, basic_agent, plan, system_message
+from inspect_ai.tool import bash, python
+
+
+@task
+def ctf(
+    challenges_dir: Path | list[Path] | None = None,
```

Comment:
In order to support retries we are going to want the task parameters to be serialisable whenever possible (this is how we can reconstruct a task from a log file). I would change the `challenges_dir` to `challenges` and make it a `str`. I would also keep the `filters` argument but provide some convenience args for serialisable versions of common filters (maybe `challenges` is the only common one though?)

Reply:
It seems like maybe `variant` would be another param to provide a convenience method for?

Reply:
Yeah this makes sense. The `Path` to `str` change will require some in-depth changes to `dataset.py` but seems good. What might the convenience methods for `filters` and `variant` look like?

Reply:
It won't be a method, it will just be two liner where if someone e.g. passes `variant` you just filter for them based on the variant.

Reply:
That's great to know, we'll make the parameters simple `str` or `list[str]`. For the `agent` parameter which is `Solver` - will inspect automatically be able to handle that wrt serialisation? Do we need to make sure we're registering it with `@solver`?